R/zchunk_LA118.hydro.R

```diff
@@ -29,41 +29,177 @@ module_energy_LA118.hydro_DISABLED <- function(command, ...) {
+      .$value -> # Convert table to single number
+      Hydro_tech_econ
+
+    # For countries with technical potential reported but no economic potential, estimate the economic potential
+    Hydropower_potential %>%
+      mutate(Economic_GWh = if_else(is.na(Economic_GWh), Technical_GWh * Hydro_tech_econ, Economic_GWh)) %>%
+      # This still leaves a few countries for which no estimates of hydro potentials are provided.
+      # The largest among them is North Korea; most of the others are islands or deserts that probably don't have much beyond present-day production.
+      # Not worrying about these (future growth rates will be set to 0)
+      mutate(Economic_EJ = Economic_GWh * CONV_GWH_EJ) %>% # Convert GWh to EJ
+      select(iso, Economic_EJ) -> # Stripping down unneeded information from table
+      Hydropower_potential
+
+    # Calculate the growth potential by country, as the economic potential minus the actual generation in the most recent historical year (from the IEA balances)
+      L100.IEA_en_bal_ctry_hist %>%
+        filter(FLOW == "ELOUTPUT", PRODUCT == "Hydro", year == 2010) %>%
+        mutate(value = value * CONV_GWH_EJ) %>%
+        left_join_error_no_match(iso_GCAM_regID, by = "iso") %>%
+        select(iso, region_GCAM3, year, value) ->
+        L118.out_EJ_ctry_elec_hydro_fby
+
+      L118.out_EJ_ctry_elec_hydro_fby %>%
+        group_by(region_GCAM3, year) %>%
+        summarise(value = sum(value)) %>%
+        ungroup() ->
+        L118.out_EJ_RG3_elec_hydro_fby
+
+    # Calculate the future growth potential in each country as the economic potential minus the present-day generation
+    Hydropower_potential %>%
+      left_join(L118.out_EJ_ctry_elec_hydro_fby, by = "iso") %>%
+      mutate(Growth_potential_EJ = Economic_EJ - value) %>%
+      mutate(Growth_potential_EJ = if_else(is.na(Growth_potential_EJ) | Growth_potential_EJ < 0, 0, Growth_potential_EJ)) %>%
+      select(-region_GCAM3, -Economic_EJ, -value) %>%
+      left_join(
+        select(iso_GCAM_regID, -country_name, -GCAM_region_ID), # Some countries (e.g., Bostwana) have NAs for RG3 names. This step is updating them (except Kosovo).
+        by = "iso") ->
+      Hydropower_potential
+
+    # Downscaling of GCAM 3.0 scenario to the country level: Assign future growth according to shares of growth potential by country
+    # Interpolate any between years, and use the final year as a proxy for years thereafter
+    # Drop 2020 because it is lower than 2010 in many regions
+    L118.out_EJ_RG3_elec_hydro_fby %>%
+      spread(year, value) %>%
+      left_join_error_no_match(A18.hydro_output, by = "region_GCAM3") %>%
+      gather(year, value, -region_GCAM3) %>%
+      mutate(year = as.numeric(year)) %>%
+      filter(year != 2020) -> # Dropping 2020, as described above
+      L118.out_EJ_RG3_elec_hydro_Y_with_values
+
+    CUSTOM_YEARS <- seq(2010, 2100, 5) # Creating custom year vector to create table in next step
```

Comment:
Previous three lines: we'd really like to not have hard-coded dates like this. What is CUSTOM_YEARS, anyway? Also re dropping 2010 because lower than 2010, perhaps open an issue to follow up on this...seems a very ad hoc decision.